viewing_party/party.py

```diff
@@ -1,23 +1,162 @@
 # ------------- WAVE 1 --------------------
+from platform import libc_ver
+
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title != None:
+        new_movie["title"] = title
+        if genre != None:
+            new_movie["genre"] = genre
+            if rating != None:
+                new_movie["rating"] = rating
+                return new_movie
+            else:
+                return None
+        else:
+            return None
+    else:
+        return None
+            
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][i]["title"]: 
+            user_data["watched"].append(user_data["watchlist"][i])
+            user_data["watchlist"].pop(i)
+            
+            return user_data

```

Comment:
While using `range` and `pop` here is fine, you can definitely make this a bit cleaner and easier to read by using the `for` loop to iterate through the movies in `user_data["watchlist"]` directly, though you'll need to replace the `pop` call with `remove`:

```python
for movie in user_data["watchlist"]:
    if title == movie["title"]:
        user_data["watched"].append(movie)
        user_data["watchlist"].remove(movie)
        return user_data
```

Reply:
However, good job at avoiding the issue of modifying the list you are iterating through by returning early on line 36!

Reply:
There are other loops that could be cleaned up by using `for` like this rather than `range` but I won't point them all out in the goal of avoiding repeating myself.